lib/ecto/changeset.ex

```diff
@@ -767,8 +777,27 @@ defmodule Ecto.Changeset do
       using `with: {Author, :special_changeset, ["hello"]}` will be invoked as
       `Author.special_changeset(changeset, params, "hello")`
   """
+  #@deprecated "use cast instead"
```

Comment:
It throws warnings in tests if it is tagged as deprecated. I haven't yet determined the right strategy to test both the legacy method and the new method for all tests, so I still use the cast_embed in tests for now.

Reply:
Ah, got it - that's compile time warning and there is really simple way to avoid such warnings:

```elixir
# in test body instead of:
My.deprecated(:call)
# use:
module = My
# module here is runtime variable!
module.deprecated(:call)
```

I remember this perfectly as I had exactly same problem with `Enum.chunk/3` and `Enum.chunk/4`:
https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/enum_test.exs#L50-L53